MediaBrowser.Providers/MediaInfo/AudioResolver.cs

```diff
@@ -0,0 +1,164 @@
+#pragma warning disable CA1002, CS1591
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Emby.Naming.Audio;
+using Emby.Naming.Common;
+using Jellyfin.Extensions;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.MediaEncoding;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Dlna;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.MediaInfo;
+
+namespace MediaBrowser.Providers.MediaInfo
+{
+    public class AudioResolver
+    {
+        public async Task<List<MediaStream>> GetExternalAudioStreams(
+            Video video,
+            int startIndex,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache,
+            ILocalizationManager localizationManager,
+            IMediaEncoder mediaEncoder,
+            CancellationToken cancellationToken)
+        {
+            var streams = new List<MediaStream>();
+
+            if (!video.IsFileProtocol)
+            {
+                return streams;
+            }
+
+            List<string> paths = GetExternalAudioFiles(video, directoryService, namingOptions, clearCache);
+
+            await AddExternalAudioStreams(streams, paths, startIndex, localizationManager, mediaEncoder, cancellationToken);
+
+            return streams;
+        }
+
+        public List<string> GetExternalAudioFiles(
+            Video video,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache)
+        {
+            List<string> paths = new List<string>();
+
+            if (!video.IsFileProtocol)
+            {
+                return paths;
+            }
+
+            paths.AddRange(GetAudioFilesFromFolder(video.ContainingFolderPath, video.Path, directoryService, namingOptions, clearCache));
+            paths.AddRange(GetAudioFilesFromFolder(video.GetInternalMetadataPath(), video.Path, directoryService, namingOptions, clearCache));
```

Comment:
It's very unlikely that the audio files are in the internal metadata folder. Subtitles are a special case because the subtitle downloader plugins can opt to save them in there, but no such thing exists for audio files.
```suggestion
```